internal/utils/yaml/yaml.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package yaml
+
+import (
+	"reflect"
+
+	"sigs.k8s.io/yaml"
+)
+
+// MergeYAML merges two yaml files. The second yaml file will override the first one if the same key exists.
+// This method can add or override a value within a map, or add a new value to a list.
+// Please note that this method can't override a value within a list.
+func MergeYAML(base, override string) (string, error) {
```

Comment:
can we use https://github.com/evanphx/json-patch instead, i.e. convert the YAMLs into JSON using `sigs.k8s.io/yaml` and then use `jsonpatch.MergePatch` and convert it back to YAML ?

Reply:
`jsonpatch.MergePatch` can't be used to merge two JSON strings, and there are no other methods to merge JSON strings either.

See https://github.com/evanphx/json-patch#create-and-apply-a-merge-patch